clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -74,17 +102,30 @@ public String name() {
      * Record an occurrence, this is just short-hand for {@link #record(double) record(1.0)}
      */
     public void record() {
-        record(1.0);
+        if (maybeRecord()) {
+            record(1.0);
+        }
     }
 
     /**
+     * @return true if the sensor's record level indicates that the metric will be recorded, false otherwise
+     */
+    public boolean maybeRecord() {
+        if (this.recordLevel.ordinal() <= config.recordLevel().ordinal()) {
```

Comment:
We shouldn't really be using the `ordinal` in this way IMO. We should expose a method in the `RecordLevel` class to return a boolean in this case. For example, the invocation could look like `recordLevel.shouldRecord(config.recordLevel)`.

Also, using the `ordinal` internally can work, but it's a bit opaque (i.e. if people change the order of definition, they break the code). One often adds a new parameter to the enum instance to make it clearer.